### PR TITLE
create api gateway test

### PR DESCRIPTION
**Pull request by wconti27**
## Motivation

tests api gateway inferred span creation

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: utils/_features.py [hunk 1]
```diff
@@ -2426,3 +2426,13 @@ def serverless_span_pointers(test_object):
...
         return test_object
+    
+    @staticmethod
+    def aws_api_gateway_inferred_span_creation(test_object):
+        """
+        AWS Api Gateway: Tests that API Gateway inferred span creation works as expected
+
+        https://feature-parity.us1.prod.dog/#/?feature=329
+        """
+        pytest.mark.features(feature_id=329)(test_object)
```
Comment (robertomonteromiguel) [inline utils/_features.py:2437]: Checking this on FPD I can see the feature 329 is "Attach stack trace to IAST vulnerabilities"

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
- lint failures
- Check the FPD feature id

File: utils/_features.py (hunk 1 above)
Comment (wconti27) [reply utils/_features.py]: fixed

**Review by robertomonteromiguel — CHANGES_REQUESTED** [review]
There are conflicts with main branch.
are we testing this new test for any language?

**Comment by wconti27** [comment]
> There are conflicts with main branch. are we testing this new test for any language?

@robertomonteromiguel I will fix the conflicts, this feature will be added to several tracers this quarter, the first being the node tracer (the PR is still not merged, probably the feature will be merged by EOD today in node).

File: utils/build/docker/nodejs/express4/integrations/api_gateway.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+function initRoutes (app, tracer) {
+  console.log('Api Gateway routes initialized.')
+
+  app.get('/api-gateway/span-creation', (req, res) => {
```
Comment (cbeauchesne) [inline utils/build/docker/nodejs/express4/integrations/api_gateway.js:4]: Could you document in `docs/weblog` what this endpoint is supposed to do ?

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: utils/_features.py [hunk 3]
```diff
@@ -2427,6 +2427,15 @@ def serverless_span_pointers(test_object):
...
         return test_object
 
+    @staticmethod
+    def aws_api_gateway_inferred_span_creation(test_object):
+        """
+        AWS Api Gateway: Tests that API Gateway inferred span creation works as expected
+
+        https://feature-parity.us1.prod.dog/#/?feature=341
+        """
+        pytest.mark.features(feature_id=341)(test_object)
```
Comment (cbeauchesne) [inline utils/_features.py:2437]:
```suggestion
        pytest.mark.features(feature_id=341)(test_object)
        
        return test_object
```

File: utils/build/docker/nodejs/express4/integrations/api_gateway.js (hunk 2 above)
Comment (wconti27) [reply utils/build/docker/nodejs/express4/integrations/api_gateway.js]: done

**Review by robertomonteromiguel — APPROVED** [review]

**Review by cbeauchesne — APPROVED** [review]

**Review by zarirhamza — APPROVED** [review]

**Pull request merged**